### PR TITLE
Add script to run a semicolon-delimited list of functional test scripts

### PR DESCRIPTION
**Pull request by scottbommarito**
Will configure VSTS to call this script shortly after this merges.

https://github.com/NuGet/Engineering/issues/316

### CODE REVIEW

File: tests/Scripts/RunTestScripts.ps1 [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+$dividerSymbol = "~"
+
+$failedTests = New-Object System.Collections.ArrayList
+
+$TestCategories.Split(';') | ForEach-Object {
+    Write-Host ($dividerSymbol * 20)
+    Write-Host "Testing $_."
+    Write-Host ($dividerSymbol * 10)
+    
+    & cmd /c "$PSScriptRoot\Run$_.bat"
```
Comment (agr) [inline tests/Scripts/RunTestScripts.ps1]: `$env:COMSPEC` instead of `cmd` maybe?

Comment (scottbommarito) [inline tests/Scripts/RunTestScripts.ps1]: wow i didn't know that existed, awesome!

File: tests/Scripts/RunTestScripts.ps1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+}
+
+Write-Host ($dividerSymbol * 20)
+if ($failedTests.Count -gt 0) {
+    Write-Host "Some functional tests failed!"
+    
+    $failedTestsStrings = $failedTests | ForEach-Object { $_ }
+    $failedTestsString = [string]::Join(", ", $failedTestsStrings)
+    Write-Host "$failedTestsString failed!"
```
Comment (agr) [inline tests/Scripts/RunTestScripts.ps1]: `Write-Host "$($failedTests -join ', ')"`? Why do you need separate copy?

File: tests/Scripts/RunTestScripts.ps1 [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        $failedTests.Add($_) | Out-Null
+    } else {
+        Write-Output "$_ succeeded!"
+    }
+}
+
+Write-Output ($dividerSymbol * 20)
+if ($failedTests.Count -gt 0) {
+    Write-Output "Some functional tests failed!"
+    Write-Output "$($failedTests -join ', ') failed!"
```
Comment (ryuyu) [inline tests/Scripts/RunTestScripts.ps1]: How noisy is it if we put each failed test on a new line?
If a bunch of tests fail, comma joined is a bit hard to read for me.

Comment (scottbommarito) [inline tests/Scripts/RunTestScripts.ps1]: It only shows the test suites as a whole, aka P0Tests, P1Tests, P2Tests, so it's not too noisy. I'll switch it to what you suggest.

**Pull request merged**